awx/main/dispatch/worker/callback.py

```diff
@@ -26,7 +30,37 @@ class CallbackBrokerWorker(BaseWorker):
 
     MAX_RETRIES = 2
 
+    def __init__(self):
+        self.buff = {}
+        self.last_write = tz_now()
+
+    def read(self, queue):
+        try:
+            return queue.get(block=True, timeout=.25)
+        except QueueEmpty:
+            return 'FLUSH'
+
+    def flush(self):
+        should_flush = (
+            any([len(events) == 2000 for events in self.buff.values()]) or
```

Comment:
2000 is probably overkill - I doubt we'd see this many events per quarter second per worker process from a real Ansible playbook.